### PR TITLE
Add events handlers

### PR DESCRIPTION
**Pull request by GigaXenuM**

### CODE REVIEW

File: app/src/application.cpp [hunk 1]
```diff
@@ -7,15 +7,15 @@
 
 #include "tools/color.h"
 
-Application::Application(MainWindow *window) : _window{ window }
```
Comment (NoName9328) [inline app/src/application.cpp:10]: why we not use pointer* ? (_window)

File: app/src/event/event.h [hunk 2]
```diff
@@ -0,0 +1,19 @@
+#pragma once
+
+class Event
+{
+public:
+    enum class EventType
+    {
```
Comment (NoName9328) [inline app/src/event/event.h:7]: please add "undefined" enum for case when we don't know what button was toggled

File: app/src/event/keyevents/keypressevent.h [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+#include "event/event.h"
+
+#include "keyboard/keyboard.h"
+
+class KeyPressEvent : public Event
+{
+public:
+    explicit KeyPressEvent(Keyboard::Key key, Keyboard::Modifier modifier);
+
+    EventType type() const override
```
Comment (NoName9328) [inline app/src/event/keyevents/keypressevent.h:12]: please move definition for this method to source file

File: app/src/scene/camera/camera.cpp [hunk 4]
```diff
@@ -50,10 +59,72 @@ const glm::mat4 &Scene::Camera::view() const
...
+    case Event::EventType::KEY_PRESS:
+        keyPressEvent(static_cast<KeyPressEvent *>(event));
+        return;
+    case Event::EventType::KEY_RELEASE:
+        keyReleaseEvent(static_cast<KeyReleaseEvent *>(event));
+        return;
+    case Event::EventType::MOUSE_MOVE:
+        mouseMoveEvent(static_cast<MouseMoveEvent *>(event));
+        return;
+    default:
```
Comment (NoName9328) [inline app/src/scene/camera/camera.cpp:76]: please describe  default case here, we must have default behavior

File: app/src/scene/camera/camera.cpp [hunk 5]
```diff
@@ -50,10 +59,72 @@ const glm::mat4 &Scene::Camera::view() const
...
     _cameraFront = glm::normalize(_position - _direction);
     _cameraRight = glm::normalize(glm::cross(_up, _cameraFront));
     _cameraUp = glm::cross(_cameraFront, _cameraRight);
     _view = glm::lookAt(_position, _position + _direction, _cameraUp);
 }
+
+void Scene::Camera::move()
+{
+    if (_moveFront)
+        _position.x += 0.001;
```
Comment (NoName9328) [inline app/src/scene/camera/camera.cpp]: please create const variable for value

File: app/src/scene/camera/camera.h [hunk 6]
```diff
@@ -38,5 +55,10 @@ class Camera
 
     glm::mat4 _projection;
     glm::mat4 _view;
+
+    bool _moveFront{ false };
```
Comment (NoName9328) [inline app/src/scene/camera/camera.h]: may be make sense use enum class for variables below

File: app/src/scene/camera/camera.cpp (hunk 5 above)
Comment (GigaXenuM) [reply app/src/scene/camera/camera.cpp]: Its the test commit

File: app/src/scene/camera/camera.cpp (hunk 4 above)
Comment (GigaXenuM) [reply app/src/scene/camera/camera.cpp:76]: We don't need the default behavior cause the class of Camera is the last of the events handlers

File: app/src/event/event.h (hunk 2 above)
Comment (GigaXenuM) [reply app/src/event/event.h:7]: We never have an undefined event

**Pull request merged**